src/main/scala/org/broadinstitute/dsde/firecloud/service/MethodConfigurationService.scala

```diff
@@ -45,69 +42,72 @@ object MethodConfigurationService {
 
 }
 
-trait MethodConfigurationService extends HttpService with PerRequestCreator with FireCloudDirectives {
+trait MethodConfigurationService extends FireCloudDirectives with SprayJsonSupport with StandardUserInfoDirectives {
 
   private final val ApiPrefix = "workspaces"
   lazy val log = LoggerFactory.getLogger(getClass)
 
-  val routes: Route =
+  val methodConfigurationRoutes: Route = requireUserInfo() { userInfo =>
     path("template") {
       passthrough(MethodConfigurationService.remoteTemplateURL, HttpMethods.POST)
     } ~
-    path("inputsOutputs") {
-      passthrough(MethodConfigurationService.remoteInputsOutputsURL, HttpMethods.POST)
-    } ~
-    pathPrefix(ApiPrefix) {
-      pathPrefix(Segment / Segment / "method_configs") { (workspaceNamespace, workspaceName) =>
-        path("copyFromMethodRepo") {
-          post {
-            entity(as[CopyConfigurationIngest]) { ingest => requestContext =>
-              val copyMethodConfig = new MethodConfigurationCopy(
-                methodRepoName = ingest.configurationName,
-                methodRepoNamespace = ingest.configurationNamespace,
-                methodRepoSnapshotId = ingest.configurationSnapshotId,
-                destination = Option(MethodConfigurationId(
-                  name = ingest.destinationName,
-                  namespace = ingest.destinationNamespace,
-                  workspaceName = Option(WorkspaceName(
-                    namespace = workspaceNamespace,
-                    name = workspaceName)))))
-              val extReq = Post(MethodConfigurationService.remoteCopyFromMethodRepoConfigUrl, copyMethodConfig)
-              externalHttpPerRequest(requestContext, extReq)
+      path("inputsOutputs") {
+        passthrough(MethodConfigurationService.remoteInputsOutputsURL, HttpMethods.POST)
+      } ~
+      pathPrefix(ApiPrefix) {
+        pathPrefix(Segment / Segment / "method_configs") { (workspaceNamespace, workspaceName) =>
+          path("copyFromMethodRepo") {
+            post {
+              entity(as[CopyConfigurationIngest]) { ingest =>
+                val copyMethodConfig = new MethodConfigurationCopy(
+                  methodRepoName = ingest.configurationName,
+                  methodRepoNamespace = ingest.configurationNamespace,
+                  methodRepoSnapshotId = ingest.configurationSnapshotId,
+                  destination = Option(MethodConfigurationId(
+                    name = ingest.destinationName,
+                    namespace = ingest.destinationNamespace,
+                    workspaceName = Option(WorkspaceName(
+                      namespace = workspaceNamespace,
+                      name = workspaceName)))))
+                val extReq = Post(MethodConfigurationService.remoteCopyFromMethodRepoConfigUrl, copyMethodConfig)
+
+                complete { executeRequestRaw(userInfo.accessToken)(extReq) }
```

Comment:
i tried to do this in most places but looks like i missed one. i'd like to rename this trait to `MethodConfigurationApiService` and lighten up the routes a lot, pushing this logic down to a `MethodConfigurationService` and underlying `AgoraDAO`